api/swagger.yaml

```diff
@@ -4024,73 +4024,71 @@ definitions:
       Warning: "unable to pin image doesnotexist:latest to digest: image library/doesnotexist:latest not found"
 
   ContainerSummary:
-    type: "array"
-    items:
```

Comment:
Looks like this definition is referenced in two places in the swagger;

- `/system/df`  -> `responses.200.Containers`  which uses it as items in an array; so the swagger incorrectly defined it as an "array of arrays"
- `/containers/json` -> `responses -> 200` which used it directly, so an "array of objects"

I think your change makes sense; i.e. change `ContainerSummary` to an object (not an array of objects), so to make swagger validation pass, the `/containers/json` response needs to be updated.

If you amend the third commit ("Fix ContainerSummary swagger docs ") with the patch below, swagger validation should pass:

```patch
diff --git a/api/swagger.yaml b/api/swagger.yaml
index fa71b38f09..efa2b81587 100644
--- a/api/swagger.yaml
+++ b/api/swagger.yaml
@@ -5261,7 +5261,9 @@ paths:
         200:
           description: "no error"
           schema:
-            $ref: "#/definitions/ContainerSummary"
+            type: "array"
+            items:
+              $ref: "#/definitions/ContainerSummary"
           examples:
             application/json:
               - Id: "8dfafdbc3a40"
```

Reply:
Thanks for your suggestion, looks perfect. I just applied your patch, let's see what the CI will find :)